src/net/listener.rs

```diff
@@ -0,0 +1,59 @@
+use std::{io, net::SocketAddr, sync::Arc};
+
+use tokio::sync::Notify;
+
+use crate::world::World;
+
+use super::Stream;
+
+/// A simulated socket server, listening for connections.
+///
+/// All methods must be called from a host within a Turmoil simulation.
+pub struct Listener {
```

Comment:
Do we want to call this `TcpListener`? I know it doesn't *actually* use TCP, but it is meant to follow Tokio's Tcp* type semantics.

Reply:
I'm open to either. What's missing in turmoil is guidance or an actual abstraction to bind to either implementation. I'm taking the stance that it's up to the consuming application to define that, because they might have specific requirements on things like how to bind.

@marcbowes and I briefly discussed adding an example that contains an example client/server with a way to choose either turmoil or tokio::net.